nixd/lib/Sema/Lowering.cpp

```diff
@@ -381,6 +381,52 @@ void ExprAttrsBuilder::addInherited(const syntax::InheritedAttribute &IA) {
   }
 }
 
+nix::Expr *Lowering::lowerOp(const syntax::Node *Op) {
+  if (!Op)
+    return nullptr;
+
+  auto MkBinaryCall = [this](nix::PosIdx Begin, syntax::Node *LHS,
+                             syntax::Node *RHS,
+                             const std::string &Name) -> nix::ExprCall * {
+    auto *Fn = new nix::ExprVar(STable.create(Name));
+    Ctx.Pool.record(Fn);
+    auto *Ret = new nix::ExprCall(Begin, Fn, {lower(LHS), lower(RHS)});
+    return Ctx.Pool.record(Ret);
+  };
+  switch (Op->getKind()) {
+  case Node::NK_OpNot: {
+    const auto *OpNot = dynamic_cast<const syntax::OpNot *>(Op);
+    return Ctx.Pool.record(new nix::ExprOpNot(lower(OpNot->Body)));
+  }
+  case Node::NK_OpNegate: {
+    const auto *OpNegate = dynamic_cast<const syntax::OpNegate *>(Op);
+    auto *NixExprVarSub =
+        Ctx.Pool.record(new nix::ExprVar(STable.create("__sub"))); // TODO: Pos
+    return Ctx.Pool.record(new nix::ExprCall(
+        OpNegate->Range.Begin, NixExprVarSub, {lower(OpNegate->Body)}));
+  }
+  case Node::NK_OpEq: {
+    const auto *OpEq = dynamic_cast<const syntax::OpEq *>(Op);
+    return Ctx.Pool.record( // TODO: Pos
```

Comment:
actually a "bug" in the parser library, maybe we should consider the operator is a Node also, or just discard this info.